app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarMenuController.kt

```diff
@@ -224,6 +224,12 @@ class DefaultBrowserToolbarMenuController(
                     BrowserFragmentDirections.actionBrowserFragmentToSyncedTabsFragment()
                 )
             }
+            is ToolbarMenu.Item.SyncAccount -> browserAnimator.captureEngineViewAndDrawStatically {
```

Comment:
Why do we need to call this method? It seems to be used for only some menu items, but I can't decipher the reasoning for the others.

Reply:
I wasn't sure the original intention 🤔 do you think I should keep it as a normal `nav`?

Reply:
No idea, but let's find out so that we can be sure!

Looking at the docs for the method, it says:
>     /**
>      * Makes the swipeRefresh background a screenshot of the engineView in its current state.
>      * This allows us to "animate" the engineView.
>      */

That sounds rather confusing and unhelpful; we should update this.

Let's look at [the code](https://github.com/mozilla-mobile/fenix/blob/b3f5e557cce658abb20e482e85c755abd544f1b2/app/src/main/java/org/mozilla/fenix/browser/BrowserAnimator.kt#L48-L72) instead.

From what I can read, we're taking the EngineView and retrieving a thumbnail image from it to set as the background for the swipe-to-refresh layout.

So it seems like the idea of it is to show a thumbnail of the page, during an animation, when we navigate away from the browser.

Playing around with this code, I removed the method around settings to see what it looks like without this change compared to 'Addons' which also has the method wrapper:


https://user-images.githubusercontent.com/1370580/114457412-ffe76600-9bab-11eb-8a8f-1a1b50a80695.mp4

What I noticed is that with or without this change we see no change in the animations! My guess is, by also looking at the git history, this animator was used at a time when we did have animation and that bit of code was removed, where as this bit was not.


To answer your question now that we know what the is, let's keep the method wrapper for consistency and file an issue for removing this animator because it does nothing, as far as I can see. 

^ Hope this helps in the future to break down problems. 🙂

Reply:
I would be comfortable moving forward without the usage of the capture here and just do a simple navigation. We can investigate the state of the animation and its existing code to determine its fate in the near future.

Reply:
It's probably safer to keep it around. We're hoping to cut over a release soon-ish and it wouldn't be great to introduce a regression that we may not realize exists from just a short investigation.

This seems to have been written as a fix for the "white flashing" in dark mode before GV introduced a fix. Removing all these references together, in this case, is probably fitting so we can delete the entire `BrowserAnimator` class and work on it from scratch when we re-enable animations.